### ISSUE TITLE
TypeHintMapping does not cope with definitions containing classes with parameter placeholders

### ISSUE THREAD

**Issue by maryo** [thephpleague/tactician-bundle#75]
```xml
<service id="foo_handler" class="%foo_handler.class%">
    <tag name="tactician.handler" typehints="true"/>
</service>
```

```
[ReflectionException]
    Class %foo_handler.class% does not exist
```

**Comment by rosstuck** [comment]
Hmm, unexpected. Good catch!

I'm not sure why anyone would want this but maybe we can check that it begins and ends with a % sign, then optionally resolve that with a getParameter call. I'll take a look within the next couple days or let me know if you want to give it a shot as a PR! :)

**Comment by rosstuck** [comment]
We'll probably need to change TagBasedMapping to pass the ContainerBuilder in as well as the Definition but that's alright with me. :)

**Comment by chalasr** [comment]
See #76

**Comment by rosstuck** [comment]
Totally resolved by @chalasr in #76 (thanks!) but please  reopen this issue if you find any other edge cases, @maryo. Thanks!
